src/shared/lib/URLWrapper.ts

```diff
@@ -1,23 +1,30 @@
-import {autorun, computed, extendObservable, intercept, IObservableObject, observable} from "mobx";
+import {autorun, computed, extendObservable, intercept, IObservableObject, IReactionDisposer, observable} from "mobx";
 import ExtendedRouterStore from "./ExtendedRouterStore";
+import hashString from "shared/lib/hashString";
+import * as _ from "lodash";
 
 export type Property<T> = {
     name: keyof T,
-    isSessionProp: boolean
+    isSessionProp: boolean,
+    aliases?: string[],
 };
 
-export default class URLWrapper<QueryParamsType> {
-    public query:Partial<QueryParamsType>;
+
+
+export default class URLWrapper<QueryParamsType extends { [key:string] : string | undefined }> {
+    public query:QueryParamsType;
+    public reactionDisposer: IReactionDisposer;
+    protected pathContext:string;
 
     constructor(
         protected routing:ExtendedRouterStore,
         protected properties:Property<QueryParamsType>[]
     ) {
         const initValues:Partial<QueryParamsType> = {};
         for (const property of properties) {
-            initValues[property.name] = undefined;
+            initValues[property.name] = (routing.query as QueryParamsType)[property.name];
```

Comment:
as a reminder, we realized we could do this up front instead of starting at undefined